cedar-policy-core/src/entities/conformance.rs

```diff
@@ -179,42 +170,78 @@ pub fn typecheck_value_against_schematype(
     }
 }
 
-/// Check whether the given `RestrictedExpr` is a valid instance of `SchemaType`
-pub fn does_restricted_expr_implement_schematype(
+/// Check whether the given `RestrictedExpr` is a valid instance of
+/// `SchemaType`.  We do not have type information for unknowns, so this
+/// function liberally treats unknowns as implementing any schema type.
+fn does_restricted_expr_implement_schematype(
     expr: BorrowedRestrictedExpr<'_>,
     expected_ty: &SchemaType,
-) -> bool {
+    extensions: &Extensions<'_>,
+) -> Result<bool, ExtensionFunctionLookupError> {
     use SchemaType::*;
 
-    match expected_ty {
+    /// Returns `Ok(true)` only when all elements are `Ok(true)`. Return an error
+    /// result if any elements are `Err(_)`. Otherwise returns `Ok(false)`.
+    fn try_all<E>(i: impl Iterator<Item = Result<bool, E>>) -> Result<bool, E> {
+        Ok(i.collect::<Result<Vec<_>, _>>()?.iter().all(|b| *b))
+    }
+
+    // Check for `unknowns`.  In this case we just don't have the information to
+    // know whether the attribute value (an unknown) matches the expected type.
+    // For now we consider this as passing -- we can't really report a type
+    // error <https://github.com/cedar-policy/cedar/issues/418>.
```

Comment:
This comment is confusing in light of the fact that the code below actually does check `u.type_annotation` if it is present.  Maybe the comment belongs in the `None` branch, ie when no type annotation is present?